packages/transformer/src/IModelTransformer.ts

```diff
@@ -1994,20 +2014,36 @@ export class IModelTransformer extends IModelExportHandler {
     // It is possible and apparently occasionally sensical to delete a model without deleting its underlying element.
     // - If only the model is deleted, [[initFromExternalSourceAspects]] will have already remapped the underlying element since it still exists.
     // - If both were deleted, [[remapDeletedSourceEntities]] will find and remap the deleted element making this operation valid
+    let sql: string;
+    if (this.hasDefinitionContainerDeletionFeature) {
+      sql = `
+      SELECT 1
+      FROM bis.DefinitionPartition
+      WHERE ECInstanceId=?
+      UNION
+      SELECT 1
+      FROM bis.DefinitionContainer
+      WHERE ECInstanceId=?
+    `;
+    } else {
+      sql = `
+      SELECT 1
+      FROM bis.DefinitionPartition
+      WHERE ECInstanceId=?
+    `;
+    }
     const targetModelId: Id64String =
       this.context.findTargetElementId(sourceModelId);
 
     if (!Id64.isValidId64(targetModelId)) return;
 
     if (this.exporter.sourceDbChanges?.element.deleteIds.has(sourceModelId)) {
       const isDefinitionPartition = this.targetDb.withPreparedStatement(
-        `
-        SELECT 1
-        FROM bis.DefinitionPartition
-        WHERE ECInstanceId=?
-      `,
+        sql,
         (stmt) => {
           stmt.bindId(1, targetModelId);
+          if (this.hasDefinitionContainerDeletionFeature)
```

Comment:
Since same id is bound twice, I suggest to use named parameter in your EC SQL instead:
```SQL
SELECT 1
FROM bis.DefinitionPartition
WHERE ECInstanceId=:targetModelId
UNION
SELECT 1
FROM bis.DefinitionContainer
WHERE ECInstanceId=:targetModelId
```
With named parameter you will not need `if` statement:
```typescript
stmt.bindId("targetModelId", targetModelId);
```